sa/saro.go

```diff
@@ -884,32 +893,29 @@ func (ssa *SQLStorageAuthorityRO) GetValidOrderAuthorizations2(ctx context.Conte
 		&ams,
 		fmt.Sprintf(`SELECT %s FROM authz2
 			LEFT JOIN orderToAuthz2 ON authz2.ID = orderToAuthz2.authzID
-			WHERE authz2.registrationID = :regID AND
-			authz2.expires > :expires AND
-			authz2.status = :status AND
-			orderToAuthz2.orderID = :orderID`,
+			WHERE orderToAuthz2.orderID = :orderID`,
```

Comment:
Fairly certain this can just be:

```go
	var ams []authzModel
	_, err := ssa.dbReadOnlyMap.Select(
		ctx,
		&ams,
		fmt.Sprintf(`SELECT %s FROM authz2
			LEFT JOIN orderToAuthz2 ON authz2.ID = orderToAuthz2.authzID
			WHERE orderToAuthz2.orderID = ?`,
			strings.Join(qualifiedAuthzFields, " "),
		),
		req.Id,
	)
```

Reply:
Yes, but questionmark placeholders are not compatible with postgres (it uses $N placeholders), so it would feel like a step backwards to replace this named placeholder (which is compatible with both mariadb and proxysql) with a question mark.